### ISSUE TITLE
[Bug]: Cannot create full width/height window under specific conditions [windows]

### ISSUE THREAD

**Issue by carterbs** [electron/electron#32238]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a feature request that matches the one I want to file, without success.

### Electron Version

13.1.6

### What operating system are you using?

Windows

### Operating System Version

Windows 10 Pro version 20H2, OS build 19042.746

### What arch are you using?

x64

### Expected Behavior

GIVEN the user has a 4k monitor and a 1080p monitor
AND the 1080p monitor is set as the 'main' display
AND force-device-scale-factor is 1
WHEN the user calls `new BrowserWindow` with full width and height bounds
THEN the created window takes up full height and width

### Actual Behavior

The created window ends up being 1920 x 1040 instead of 3840 x 2120.

Logs below:

![image](https://user-images.githubusercontent.com/16466430/146576539-a28843c8-5175-4691-bc28-a09ea6e50c48.png)

### Testcase Gist URL

https://gist.github.com/415e8aac45a9569d560ccf1f459e0614

### Additional Information

Here's the monitor layout:
3 and 2 are 1080p. 1 is 4k. 1 is scaled at 300%. The other two are 100%. 3 is the 'main' display'.

![image](https://user-images.githubusercontent.com/16466430/146575906-75488f20-cff0-4b49-b084-099322bf26da.png)

If the 4k monitor is set as 'main', the bug doesn't happen.
If you leave out the commandLine switch for device scale factor, the bug doesn't happen.
If monitor 2 is the 'main' display, the bug DOES happen.
If you swap the position of monitor 1 and monitor 2, the bug DOES happen.

**Comment by machinaeXphilip** [comment]
I have a similar issue on MacOS 11.4

It seems like the main windows width and height dimensions are limiting the maximum width and height of new BrowserWindow() on secondary screens.

Only way around is to make the main screen the one with the highest resolution.

**Comment by dsanders11** [comment]
@carterbs @machinaeXphilip - can either of you please try testing with Electron Fiddle on the newest version of Electron? The issue is reported for v13 which is no longer supported.

Sounds like it could be related to #32773.

**Comment by machinaeXphilip** [comment]
Yes. Thanks for asking again. Electron 17.0.0 fixed a lot of rendering issues, but in my case i still need to fire win.setBounds on the ready-to-show event to make sure a window on a 2nd screen can be larger than the window size of the main window.

Weirdly enough the issue does not come up when I create the window with `transparent:true`. This makes me think I might have other issues in my project. Gonna try to reproduce the issue in a fiddle with minimal setup.

**Comment by machinaeXphilip** [comment]
on mac OS 11.4  
with a second screen with 1920x1080 resolution on the left side of the main display, a MacBook Display with 1440x900, this results in an 1440x900 pixel sized window.

transparent seems to not change anything. tested also on electron 17.0.0 as well as 18.0.0-beta.2

https://gist.github.com/dd9a1056890c2ceb2222477d00fa0b90

**Comment by dsanders11** [comment]
@machinaeXphilip, thanks for confirming it's still an issue.

I've also confirmed it on macOS, in a more default setup of a second display to the right of the MBP display. I've created [this fiddle](https://gist.github.com/f1e1a6038d43b6732e11396d7412209e) to reproduce, an exit code of 0 means success, 1 means failed, 2 means prerequisites not met.

I'm fairly certain this is related to #32773, which is limiting a window size to the size of the primary display, regardless of where the window is placed. I'll leave them as separate issues for the moment, this issue points out an important distinction in multiple display environments.
